designer/client/src/components/aiAssistant/components/Composer.tsx

```diff
@@ -0,0 +1,137 @@
+    if (textarea) {
+        resetInputHeight(textarea);
+
+        // Define modal maximum height (50vh)
+        const modalMaxHeight = window.innerHeight * 0.5;
+
+        // Calculate new height and cap it if needed
+        let newHeight = textarea.scrollHeight;
+        if (newHeight > modalMaxHeight) {
+            newHeight = modalMaxHeight;
+            textarea.style.overflowY = "auto";
+        } else {
+            textarea.style.overflowY = "hidden";
+        }
+
+        textarea.style.height = `${newHeight}px`;
+
+        // Anchor the bottom by shifting the extra height upward
+        const baseHeight = 38; // adjust base height as needed
+        const diff = newHeight - baseHeight;
+        if (newHeight < modalMaxHeight) {
+            textarea.style.marginTop = `-${diff}px`;
+        }
+    }
+};
+
+export const Composer = () => {
+    const { send, setText, cancel } = useComposerRuntime();
+    const { scrollToBottom, provideBottomSpacer } = UseScrollToBottom();
+
+    const [message, setMessage] = useState("");
+    const textAreaRef = useRef<HTMLTextAreaElement>(null);
+    const { isRunning } = useThread();
+    const isSendDisabled = !message || isRunning;
+
+    const handleChange = useCallback(
+        (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+            const newValue = event.target.value;
+            setMessage(newValue);
+            setText(newValue);
+            adjustInputHeight(textAreaRef.current);
+        },
+        [setText],
+    );
+
+    const handleSend = useCallback(() => {
+        if (isSendDisabled) {
+            return;
+        }
+
+        provideBottomSpacer();
+        scrollToBottom();
+        send();
+        setMessage("");
+        resetInputHeight(textAreaRef.current);
+    }, [isSendDisabled, provideBottomSpacer, scrollToBottom, send]);
+
+    const handleCancel = useCallback(() => {
+        cancel();
+    }, [cancel]);
+
+    const handleKeyDownOnInput = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+        if (event.key === "Enter" && !event.shiftKey) {
+            event.preventDefault();
+            handleSend();
+        }
+    };
+
+    useEffect(() => {
+        scrollToBottom();
+    }, [scrollToBottom]);
+
+    return (
+        <StyledRoot data-no-focus-lock>
+            <NodeTable sx={{ margin: 0, marginRight: "16px", width: "100%" }}>
+                <div className={nodeValue}>
+                    <StyledTextArea
+                        autoFocus
+                        onFocus={(e) => {
+                            // Disable nk-windows focus event which move second window when opened, to the background on input clicks
+                            e.stopPropagation();
+                        }}
+                        ref={textAreaRef}
+                        autoComplete="off"
+                        name="message"
+                        placeholder="Message AI Assistant"
```

Comment:
Could we add example question such as: `Message AI Assistant, e.g. what is a scenario?`